result.go

```diff
@@ -31,54 +33,80 @@ type Result struct {
 	FullName      string         `json:"fullName,omitempty"`
 	Labels        []Label        `json:"labels,omitempty"`
 }
+type FailureMode string
 
 //Before defines a step
 type Before struct {
 	Name          string         `json:"name,omitempty"`
 	Status        string         `json:"status,omitempty"`
-	StatusDetails *StatusDetails `json:"statusDetails,omitempty"`
+	StatusDetails *statusDetails `json:"statusDetails,omitempty"`
 	Stage         string         `json:"stage,omitempty"`
 	Description   string         `json:"description,omitempty"`
 	Start         int64          `json:"start,omitempty"`
 	Stop          int64          `json:"stop,omitempty"`
-	Steps         []Step         `json:"steps,omitempty"`
+	Steps         []stepObject   `json:"steps,omitempty"`
 	Attachments   []Attachment   `json:"attachments,omitempty"`
 }
 
+type HasSteps interface {
+	GetSteps() []stepObject
+	AddStep(step stepObject)
+}
+
+func (r *result) GetSteps() []stepObject {
+	return r.Steps
+}
+
+func (r *result) AddStep(step stepObject) {
+	r.Steps = append(r.Steps, step)
+}
+
 type Parameter struct {
 	Name  string `json:"name,omitempty"`
 	Value string `json:"value,omitempty"`
 }
 
 var wsd, resultPath string
 
+const (
+	ALLURE_RESULTS_PATH = "ALLURE_RESULTS_PATH"
+	nodeKey             = "current_step_container"
+)
+
 //Test execute the test and creates an Allure result used by Allure reports
 func Test(t *testing.T, description string, testFunc func()) {
-	wsd = os.Getenv("ALLURE_RESULTS_PATH")
+	wsd = os.Getenv(ALLURE_RESULTS_PATH)
+	if wsd == "" {
+		log.Fatalf(fmt.Sprintf("%s environment variable cannot be empty", ALLURE_RESULTS_PATH))
+		os.Exit(1)
```

Comment:
log.Fatal will already exit - so this line is useless

Reply:
@vbehar is it a good practice for an external lib to exit?

Reply:
no, indeed. we could fallback to a path like `/tmp/allure-result` using `os.TempDir()` with logging a warning msg about the output path

Reply:
It would probably make sense to output folder location once to make it easier for the user. This way they wouldn't have to guess why they're not seeing the folder where they expect it to be. I put this check for this particular reason.